src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -45,7 +44,7 @@ protected override Task<ImmutableArray<ISymbol>> GetSymbolsWorker(SyntaxContext
                 cancellationToken).ConfigureAwait(false);
 
             // Don't preselect intrinsic type symbols so we can preselect their keywords instead.
-            return symbols.WhereAsArray(s => inferredTypes.Contains(GetSymbolType(s)) && !IsInstrinsic(s));
+            return symbols.Where(s => inferredTypes.Contains(GetSymbolType(s)) && !IsInstrinsic(s)).Select(s => (s, CompletionItemRules.Default)).ToImmutableArray();
```

Comment:
This is a fairly allocation sensitive code-path.  Consider unwinding the `Where/Select/ToImmutableArray` with a pulled array builder.